File: backends/credhub/src/test/java/org/cloudfoundry/credhub/utils/MultiJsonPathMatcher.java
package org.cloudfoundry.credhub.utils;

import org.springframework.test.web.servlet.ResultMatcher;

import com.jayway.jsonpath.JsonPath;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.MatcherAssert.assertThat;

final public class MultiJsonPathMatcher {

  private MultiJsonPathMatcher() {
    super();
  }

  public static ResultMatcher multiJsonPath(final Object... keysAndValues) {
    return result -> {
      for (int i = 0; i < keysAndValues.length; ) {
        final String jsonPath = (String) keysAndValues[i++];
        final Object expectedValue = keysAndValues[i++];

        assertThat("field " + jsonPath, JsonPath.compile(jsonPath).read(result.getResponse().getContentAsString()),
          equalTo(expectedValue));
      }
    };
  }
}

File: backends/credhub/src/test/kotlin/org/cloudfoundry/credhub/utils/MultiJsonPathMatcher.kt
package org.cloudfoundry.credhub.utils

import com.jayway.jsonpath.JsonPath
import org.hamcrest.CoreMatchers
import org.hamcrest.MatcherAssert
import org.springframework.test.web.servlet.MvcResult
import org.springframework.test.web.servlet.ResultMatcher

class MultiJsonPathMatcher {
    companion object {
        @JvmStatic
        fun multiJsonPath(vararg keysAndValues: Any): ResultMatcher =
            ResultMatcher { result: MvcResult ->
                var i = 0
                while (i < keysAndValues.size) {
                    val jsonPath = keysAndValues[i++] as String
                    val expectedValue = keysAndValues[i++]
                    MatcherAssert.assertThat(
                        "field $jsonPath",
                        JsonPath.compile(jsonPath).read(result.response.contentAsString),
                        CoreMatchers.equalTo(expectedValue),
                    )
                }
            }
    }
}
